### ISSUE TITLE
Subscription#create allows duplicate Subscriptions

### ISSUE THREAD

**Issue by JPrevost** [MITLibraries/scoap3hub#140]
The model does not prevent Subscribing to the same content multiple times (sometimes with conflicting delivery preferences).

The test `SubscriptionSpec#create does not allow duplicate Subscriptions` has been created and marked as `pendingUntilFixed`.

Proposed fix would be to consider a unique index on `subscriberId - topicId`.

**Comment by richardrodgers** [comment]
I agree that the current behavior is undesirable, but the primary reason is that the date range handling of subscriptions has not yet been implemented. When it is, I believe it would be legitimate for a subscriber to create a subscription to topic X, cancel it (or let it expire when the date passes), then resubscribe to X (with, as noted, perhaps different delivery preferences).

I think we should keep this ticket open as a reminder that that functionality has to be implemented in a way that avoids the ticket's quite valid concern:  2 or more active conflicting subscriptions to a topic

**Comment by JPrevost** [comment]
Sounds good to me.
